### PR TITLE
[react-native] Add SectionList ListEmptyComponent

### PR DESCRIPTION
**Pull request by kamal**
Added in 0.45.x. There was a similar PR that only added it to FlatList #17231.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native/commit/264d67c424841660874bcf11ad686aea57fe7a44
- [N/A] Increase the version number in the header if appropriate.
- [N/A] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by iRoachie — APPROVED** [review]

**Comment by kamal** [comment]
@iRoachie i just noticed that there was a fix made to FlatList's ListEmptyComponent type, to accept `React.ReactElement<any>`. I have made the changes to my PR. However, the other props in the same area in SectionList do not have this fix. Should I go ahead and fix them in this PR too?

**Comment by iRoachie** [comment]
@kamal It's up to you! You could change this one to more a general PR or make another one.

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@
 //                 HuHuanming <https://github.com/huhuanming>
 //                 Kyle Roach <https://github.com/iRoachie>
 //                 Tim Wang <https://github.com/timwangdev>
+//                 Kamal Mahyuddin <https://github.com/kamal>
```
Comment (alloy) [inline types/react-native/index.d.ts]: Just wanting to make sure that you understand that this means you’ll be signing up as a maintainer and will get notifications for all issues/PRs.

Same goes to you @timwangdev.

**Comment by alloy** [comment]
@kamal I’ll wait to merge until we hear back from you 👍

File: types/react-native/index.d.ts (hunk 1 above)
Comment (timwangdev) [inline types/react-native/index.d.ts]: Sure! Thanks for the notice.

**Pull request merged**